rllib/algorithms/algorithm_config.py

```diff
@@ -3935,13 +3941,13 @@ def _validate_new_api_stack_settings(self):
                 user_src = default_src
             if default_src != user_src:
                 raise ValueError(
-                    "When using the new API stack with EnvRunners, you cannot override "
-                    "the `DefaultCallbacks.on_episode_created()` method anymore! "
-                    "This particular callback is no longer supported as we are now "
-                    "using gym.vector.Env, which automatically resets individual "
-                    "sub-environments when they are terminated. Override the "
-                    "`on_episode_start` method instead, which gets fired right after "
-                    "the `env.reset()` call."
+                    "When using the new API stack in single-agent and with EnvRunners, "
```

Comment:
Yeah that's a problem I encountered when evaluating against benchmarks, i.e. running a benhcmark against the policy which means the environment needs to be reconfigured for a benchmark run. The solution is to run episodes and after x episodes switch the configuration and call the sampler again.

Reply:
Makes sense. Some users need this opportunity to send config-information to the env before(!) it is reset, but after(!) an episode ended.

We'll have to figure out a way around this for single-agent now, using vector Env. :|